netbox/utilities/testing/views.py

```diff
@@ -583,6 +598,55 @@ def test_bulk_import_objects_with_permission(self):
             self.assertHttpStatus(self.client.post(self._get_url('import'), data), 200)
             self.assertEqual(self._get_queryset().count(), initial_count + len(self.csv_data) - 1)
 
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=['*'])
+        def test_bulk_update_objects_with_permission(self):
+            if not self.csv_update_data:
+                return
+
+            data = {
+                'csv': self._get_csv_data(),
+            }
+
+            # Assign model-level permission
+            obj_perm = ObjectPermission(
+                name='Test permission',
+                actions=['add']
+            )
+            obj_perm.save()
+            obj_perm.users.add(self.user)
+            obj_perm.object_types.add(ContentType.objects.get_for_model(self.model))
+
+            # need to track ids so we know what new ids were added by csv_data so we can
+            # do the updates on the appropriate ids
+            prev_ids = list(self._get_queryset().values_list('id', flat=True).order_by('id'))
+            self.assertHttpStatus(self.client.post(self._get_url('import'), data), 200)
+            count = self._get_queryset().count()
+            new_ids = list(self._get_queryset().values_list('id', flat=True).order_by('id'))
+            diff_ids = [x for x in new_ids if x not in prev_ids]
+            start_id = diff_ids[0]
```

Comment:
I'm not comfortable baking this much logic into the test method itself. My fear is that hiding the logic responsible for retrieving the IDs of objects to be updated will inevitably lead to confusion for test authors. IMO it's much safer to specify the IDs explicitly within each test. For example, in ProvderTestCase we first create three providers:

```python
providers = (
    Provider(name='Provider 1', slug='provider-1'),
    Provider(name='Provider 2', slug='provider-2'),
    Provider(name='Provider 3', slug='provider-3'),
)
Provider.objects.bulk_create(providers)
```

We can reference these objects directly when defining the CSV update data:

```python
cls.csv_update_data = (
    "id,name,comments",
    "{providers[0].pk},Provider 7,New comment7",
    "{providers[1].pk},Provider 8,New comment8",
    "{providers[2].pk},Provider 9,New comment9",
)
```

Granted it's a bit more verbose, but with the benefit of conveying to the reader _exactly_ what data is being fed to the form without any digging.

I'm happy to make these adjustments if you're tired of dealing with the test suites at this point; I realize how mind numbing it gets. :slightly_smiling_face: